### PR TITLE
Add psp and cilium netpol

### PR DESCRIPTION
**Pull request by piontec**
This PR:
- adds a custom PSP to crossplane's `crossplane` and rbac-manager` (`emptyDir` volumes needed)
- adds CiliumNetoworkPolicy that allows to talk to api server during CRD installation Job

### CODE REVIEW

File: helm/crossplane/templates/crd-install/crd-np.yaml [hunk 1]
```diff
@@ -1,4 +1,24 @@
...
+  labels:
+    app.kubernetes.io/component: {{ include "crdInstall" . | quote }}
+    {{- include "labels.selector" . | nindent 4 }}
+    role: {{ include "crdInstallSelector" . | quote }}
+spec:
+  egress:
+  - toEntities:
+    - kube-apiserver
+  endpointSelector: {}
+{{- else }}
```
Comment (ljakimczuk) [inline helm/crossplane/templates/crd-install/crd-np.yaml:21]: Don't you need to use the `---` separator here?

File: helm/crossplane/templates/crd-install/crd-np.yaml [hunk 2]
```diff
@@ -1,4 +1,24 @@
...
+  namespace: {{ .Release.Namespace | quote }}
+  annotations:
+    # create hook dependencies in the right order
+    "helm.sh/hook-weight": "-7"
+    {{- include "crdInstallAnnotations" . | nindent 4 }}
+  labels:
+    app.kubernetes.io/component: {{ include "crdInstall" . | quote }}
+    {{- include "labels.selector" . | nindent 4 }}
+    role: {{ include "crdInstallSelector" . | quote }}
+spec:
```
Comment (ljakimczuk) [inline helm/crossplane/templates/crd-install/crd-np.yaml:16]: Spec of this policy looks the same as of the previous one `allow-apiserver-access`. Can't we rely on the previous one only?

File: helm/crossplane/templates/crd-install/crd-np.yaml (hunk 1 above)
Comment (piontec) [reply helm/crossplane/templates/crd-install/crd-np.yaml:21]: it's one or the other (if-else), it never generates 2 objects

Comment (ljakimczuk) [reply helm/crossplane/templates/crd-install/crd-np.yaml:21]: ah ok, my bad, didn't notice that

File: helm/crossplane/templates/crd-install/crd-np.yaml (hunk 2 above)
Comment (piontec) [reply helm/crossplane/templates/crd-install/crd-np.yaml:16]: Forgot to remove. They are the same currently, so we can use the one generated in pre-hook.

**Review by ljakimczuk — APPROVED** [review]

**Pull request merged**